internal/consensus/state.go

```diff
@@ -558,7 +557,7 @@ func (cs *State) updateRoundStep(round int32, step cstypes.RoundStepType) {
 }
 
 // enterNewRound(height, 0) at cs.StartTime.
-func (cs *State) scheduleRound0(rs *cstypes.RoundState) {
+func (cs *State) scheduleRound0(rs cstypes.RoundState) {
```

Comment:
So now this is a copy of it?

Reply:
Not a deep copy. Some of `RoundState` fields are pointers. Passing a `RoundState` object as value will copy _the pointers_, not whatever those pointers point to. For example:
```go
// just an example, suppose this is properly inizialized.
rs := &cstypes.RoundState{} 

cs.scheduleRound0(rs)
```
The pointer fields of the original `rs` and the `RoundState` inside the `scheduleRound0` function will point to the same data in memory.

Reply:
^

Reply:
It is a shallow copy, as commented in line 259. 

But for internal transitions like that, I am not sure whether this makes much sense.

Moreover, this method is so minor - it schedules a timeout - that this shallow copy is pretty much unnecessary.

Reply:
Done, no more shallow copy here